### PR TITLE
feat: handle code update and delete events

### PR DESCRIPTION
**Pull request by MohamadJaara**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

Conversation code update and delete events are not handled

### Solutions

map and handle the events

### Dependencies (Optional)

_If there are some other pull requests related to this one (e.g. new releases of frameworks), specify them here._

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/ConversationGroupRepository.kt [hunk 1]
```diff
@@ -280,37 +289,52 @@ internal class ConversationGroupRepositoryImpl(
...
     override suspend fun generateGuestRoomLink(conversationId: ConversationId): Either<NetworkFailure, Unit> =
         wrapApiRequest {
             conversationApi.generateGuestRoomLink(conversationId.toApi())
         }.onSuccess {
-            it.data?.let { data -> conversationDAO.updateGuestRoomLink(conversationId.toDao(), data.uri) }
-            it.uri?.let { link -> conversationDAO.updateGuestRoomLink(conversationId.toDao(), link) }
+            it.data
+            // TODO: set the correct value for is passwordProtected
+            it.data?.let { data -> conversationDAO.updateGuestRoomLink(conversationId.toDao(), data.uri, false) }
+            it.uri?.let { link -> conversationDAO.updateGuestRoomLink(conversationId.toDao(), link, false) }
```
Comment (MohamadJaara) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/ConversationGroupRepository.kt:320]: The model returned from the remote request needs to be changed and will be part of the next PR

File: logic/src/commonTest/kotlin/com/wire/kalium/logic/sync/receiver/conversation/CodeDeletedHandler.kt [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+import com.wire.kalium.logic.util.arrangement.dao.ConversionDAOArrangement
+import com.wire.kalium.logic.util.arrangement.dao.ConversionDAOArrangementImpl
+import com.wire.kalium.persistence.dao.ConversationIDEntity
+import io.mockative.eq
+import io.mockative.once
+import io.mockative.verify
+import kotlinx.coroutines.test.runTest
+import kotlin.test.Test
+
+class CodeDeletedHandlerTest {
```
Comment (alexandreferris) [inline logic/src/commonTest/kotlin/com/wire/kalium/logic/sync/receiver/conversation/CodeDeletedHandler.kt:33]: class naming is missing `Test`

File: logic/src/commonTest/kotlin/com/wire/kalium/logic/sync/receiver/conversation/CodeDeletedHandler.kt [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+    @Test
+    fun givenCodeUpdateEvent_whenhandlerIsInvoked_thenCodeIsUpdated() = runTest {
```
Comment (alexandreferris) [inline logic/src/commonTest/kotlin/com/wire/kalium/logic/sync/receiver/conversation/CodeDeletedHandler.kt:36]:
```suggestion
    fun givenCodeUpdateEvent_whenHandlerIsInvoked_thenCodeIsUpdated() = runTest {
```

File: logic/src/commonTest/kotlin/com/wire/kalium/logic/sync/receiver/conversation/CodeUpdateHandlerTest.kt [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+import io.mockative.eq
+import io.mockative.once
+import io.mockative.verify
+import kotlinx.coroutines.test.runTest
+import kotlin.test.Test
+
+class CodeUpdateHandlerTest {
+
+    @Test
+    fun givenCodeUpdateEvent_whenhandlerIsInvoked_thenCodeIsUpdated() = runTest {
```
Comment (alexandreferris) [inline logic/src/commonTest/kotlin/com/wire/kalium/logic/sync/receiver/conversation/CodeUpdateHandlerTest.kt:36]:
```suggestion
    fun givenCodeUpdateEvent_whenHandlerIsInvoked_thenCodeIsUpdated() = runTest {
```

**Review by alexandreferris — COMMENTED** [review]
Great work! Just some small comments

**Review by alexandreferris — APPROVED** [review]

**Pull request merged**